### DISCUSSION TITLE
cache-aware-clients need enhance

### DISCUSSION THREAD

**Discussion by 1257960069** [ServiceStack/Discuss#104] [Q&A]
1.
https://docs.servicestack.net/cache-aware-clients
the sample code from docs:
```
var sharedCache = new ConcurrentDictionary<string, HttpCacheEntry>();

IServiceClient msgPackClient = new MsgPackServiceClient(baseUrl).WithCache(sharedCache);

IServiceClient jsonClient = new JsonHttpClient(baseUrl).WithCache(sharedCache);
```

issue: 
how to control the sharedCache lifetime?
just work at a method body?

2.
https://docs.servicestack.net/csharp-client#cache-aware-service-clients
the sample code from docs:
```
IServiceClient client = new JsonServiceClient(baseUrl).WithCache(); 
//equivalent to:
IServiceClient client = new CachedServiceClient(new JsonServiceClient(baseUrl));
```
issue:  
WithCache can not work, because of not exist long litetime's ConcurrentDictionary<string, HttpCacheEntry>() 
by new CachedServiceClient(ServiceClientBase client)
![image](https://github.com/ServiceStack/Discuss/assets/31265244/b8ee3a2d-4e74-4bd5-8dae-9884463ea229)

**Comment by mythz** [comment]
It's not meant to be static, you can register it as a singleton to reuse the cache.

**Reply by 1257960069** [reply]
of cource, i can do it.  the root issue is that your sample code:
IServiceClient client = new CachedServiceClient(new JsonServiceClient(baseUrl));

we can not new CachedServiceClient every time
